### ISSUE TITLE
Highlight entire row, rather than a single cell if field matches string

### ISSUE THREAD

**Issue by infamousdevil** [dfinke/ImportExcel#1079]
I've been able to highlight the single cell where the 'false' is located, but can I have the entire row highlighted?
New-ConditionalText -Text "FALSE" -ConditionalTextColor black -BackgroundColor yellow -Range "B2:B1048576"

Thanks for such an amazing module!  I've been using it since 2015!

**Comment by dfinke** [comment]
Thanks! Don't know if you can do it. There is nothing on that function to let you do it explicitly. Not sure how you'd do it in Excel, maybe with a formula? You'd need to see if you can do it directly in Excel and then see if it is doable in PS.

**Comment by infamousdevil** [comment]
Doug,
Thank you for taking the time to respond!  I did find a way to do it directly in Excel: https://trumpexcel.com/highlight-rows-based-on-cell-value/
But unsure how that could be accomplished with your Export-Excel function.  Before I came across your wonderful module, I was actually using a machine with Excel installed and using something like: 
# create new Excel file$excel = New-Object -ComObject Excel.Application$excel.Visible = $true$workbook = $excel.Workbooks.Add()$workbook.ActiveSheet.Cells.Item(1,3) = 'I added this!'$ExcelPath = "$env:temp\newexcel.xlsx"$workbook.SaveAs($ExcelPath)$workbook.Close($false)$excel.Quit()
There's way more you can do with the COM Object method... it's what I was using before I came across ImportExcel.

Warm Regards,Thomas GiboneyUMtts, UpMyTech, LinkedIn
 

    On Wednesday, September 15, 2021, 02:14:05 PM CDT, Doug Finke ***@***.***> wrote:  
 
 


Thanks! Don't know if you can do it. There is nothing on that function to let you do it explicitly. Not sure how you'd do it in Excel, maybe with a formula? You'd need to see if you can do it directly in Excel and then see if it is doable in PS.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by dfinke** [comment]
Yeah, it looks like you can do it by setting a formula, I believe. The `New-ConditionalText` function does not have that. 

Yes the COM approach will have a lot more you can do, it works with a live instance of Excel. The key thing the COM approach ***cannot*** do is work without having Excel installed, like the PowerShell module :-)
